src/cephadm/cephadm

```diff
@@ -4260,8 +4260,12 @@ def _pull_image(ctx, image, insecure=False):
         if not ret:
             return
 
+        err_str = ''
+        if 'unauthorized' in err:
+            err_str = '\nCheck that correct registry credentials are provided by --registry-url, --registry-username, --registry-password, or supply --registry-json with credentials'
```

Comment:
This `_pull_image` function is run in more places than just bootstrap. For example, `_get_container_image_info` in serve.py (which is run regularly in the serve loop as well as during upgrade) calls the cephadm pull command which ends up here. This message wouldn't make sense in all of these cases. Either we need to make this message more generic or we should catch the exception raised when `_pull_image` is called during bootstrap specifically and print a message like this only there.

Reply:
It might make sense to add a more specific exception type rather than tweak the string set in Error. If you make a subclass of error, for example "class UnauthorizedRegistryError(Error)". It is then easier to catch that specific case where @adk3798 suggests above, but it will otherwise work just like Error.